Entities/Common/Controls/ImportantPickup.as

```diff
@@ -0,0 +1,102 @@
+#include "StandardControlsCommon.as"
+
+// Current baseZ ordening in the game:
+// 100: door OPEN, bridge OPEN
+// -50+: buildings
+// -40 ladders
+// relative -10: pickup behind (keg, crate)
+// -25 ballista
+// -20: fireplace
+// -10: saw (-1,-5)
+//      chest
+//      mounted bow
+// 10: carrying saw
+//     bison
+//     shark
+//     bush
+//     grain, flower
+// 20: DEFAULT IMPORTANT
+// 30: bomb arrows
+// 50: workbench foreground (weird)
+// 100: door CLOSED, bridge CLOSED
+// 500: spike
+
+
+// Add the following to the Init of the object to specify a custom hover polygon:
+// Vec2f[] hoverPoly =
+//   { Vec2f(-7.0f, 3.5f)
+//   , Vec2f(-2.5f, -5.0f)
+//   , Vec2f(1.0f, -5.0f)
+//   , Vec2f(6.5f,  3.5f)
+//   , Vec2f(-7.0f, 3.5f)
+//   };
+
+// this.set("hover-poly", hoverPoly);
+
+
+// Overwrite importance by setting "important-pickup" in the onInit
+
+void onInit(CBlob@ this)
+{
+  f32 baseZ = 20.0f;
+  if (this.exists("important-pickup")) 
+  {
+    baseZ = this.get_f32("important-pickup");
+  }
+
+  this.getSprite().SetZ(baseZ);
+  this.set_f32("important-pickup", baseZ);
+}
+
+void onAttach(CBlob@ this, CBlob@ attached, AttachmentPoint @attachedPoint)
+{
+  f32 baseZ = this.getSprite().getZ();
+  this.set_f32("important-pickup", baseZ);
+}
+
+
+void onDetach(CBlob@ this, CBlob@ detached, AttachmentPoint@ attachedPoint)
+{
+  f32 baseZ = this.get_f32("important-pickup");
+  this.getSprite().SetZ(baseZ);
+}
+
+// Add ImportantPickup.as to sprite_scripts in the .ctf file to enable rendering of 
+// the hover square (for debug purposes)
+void onRender(CSprite@ this)
```

Comment:
The debug HUD is nice for testing... Though, TBH, the script may as well add itself to the sprite and have the function early exit if `g_debug == 0`. Because right now, if one wants to check/rework the hover shapes they would need to find all blobs that use the script to edit their `.cfg` individually.

Reply:
Added a `g_debug` check